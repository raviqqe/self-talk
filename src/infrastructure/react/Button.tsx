import { defaultImport } from "default-import";
import defaultStyled from "styled-components";
import { lightGrey, red, white } from "./style/colors.js";
import { boxShadow } from "./style/index.js";

const styled = defaultImport(defaultStyled);

export const Button = styled.button<{ secondary?: boolean }>`
  ${boxShadow};
  background: ${({ secondary }) => (secondary ? lightGrey : red)};
  display: flex;
  justify-content: center;
  align-items: center;
  font-family: inherit;
  color: ${white};
  border: none;
  cursor: pointer;
  flex-shrink: 0;
`;
