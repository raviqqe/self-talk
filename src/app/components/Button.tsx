import { styled } from "@linaria/react";
import { boxShadow, lightGrey, red, white } from "../style.js";

export const Button = styled.button<{ secondary?: boolean }>`
  ${boxShadow};
  background: ${({ secondary }) => (secondary ? lightGrey : red)};
  display: flex;
  justify-content: center;
  align-items: center;
  font-family: inherit;
  color: ${white};
  border: none;
  cursor: pointer;
  flex-shrink: 0;
`;
