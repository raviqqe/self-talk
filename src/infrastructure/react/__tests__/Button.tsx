import { create } from "react-test-renderer";
import { Button } from "../Button";

it("renders", () => {
  expect(create(<Button>foo</Button>).toJSON()).toMatchSnapshot();
});

it("renders as a secondary button", () => {
  expect(
    create(<Button secondary={true}>foo</Button>).toJSON()
  ).toMatchSnapshot();
});
