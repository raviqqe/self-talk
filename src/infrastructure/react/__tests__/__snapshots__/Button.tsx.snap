// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<button
  className="sc-AxjAm fFgggO"
>
  foo
</button>
`;

exports[`renders as a secondary button 1`] = `
<button
  className="sc-AxjAm dLWtmN"
>
  foo
</button>
`;
